docs/Scriptmodule-Guide.md

```diff
@@ -0,0 +1,286 @@
+:   This property expects two parameters. One is a short handle for the license
+    used for the emulator. Examples are GPL3 (GPL v3), NONCOM
+    (Non-commercial). Define the handle UPPERCASE. The second parameter defines
+    the full text to the license applied to the emulator. Both parameters are
+    mandatory.
+
+`rp_module_repo` (usually: three mandatory parameters, fourth optional)
+
+:   This is the most complex property to fill out. The good part mainly it will
+    be `git <git-repo-url> <branch>`. But let us begin from the start. First
+    parameter is the type of source repo e.g., `git` or `svn`. Second is the URL
+    of the repo you would usually use to clone a repository. The third parameter
+    denotes the branch to be used (use `-` for Subversion based repos). The
+    fourth parameter is optional and denotes the revision to check out of that
+    branch. If no fourth parameter is given the HEAD revision will be checked
+    out. If the emulator source repo uses different branches, depending for
+    example on the architecture, you can also define a callback function. A
+    callback function starts with a colon `:` followed by the function to be
+    called. This function must be present in your scriptmodule. Use a leading
+    underscore for your function to denote that it is an internally used
+    function. You can find examples for callbacks for example in the
+    [`dolphin.sh`](https://github.com/RetroPie/RetroPie-Setup/blob/master/scriptmodules/emulators/dolphin.sh)
+    scriptmodule or
+    [`mame.sh`](https://github.com/RetroPie/RetroPie-Setup/blob/master/scriptmodules/emulators/mame.sh)
+    scriptmodule and others. You can also find the special operator `file` as
+    first parameter if you browse through the scriptmodules but this option will
+    not be covered here now.
+
+`rp_module_help` (brief text)
+
+:   Put here in quotes (`"`) a short info about where to put the ROMs, or if
+    more complex user actions are needed after installation a link to the
+    RetroPie-Docs for emulator infini. Obviously the linked documentation page
+    should exist. Newlines in the displayed text can be forced with `\n`.
+
+`rp_module_section` (predefined, pick one)
+
+:   In which section will the emulator be collated? This is relevant when the
+    user runs `retropie_setup.sh`. Most of the time you will use `exp` for
+    experimental or `opt` for optional. If an scriptmodule emulator qualifies
+    for the section `main` should be checked with the RetroPie maintainers
+    first.
+
+`rp_module_flags` (predefined, pick none to many)
+
+:   Flags, as the name suggests, control under which conditions a scriptmodule
+    can be run. Conditions may apply for the different hardwares, architectures
+    and capabilities of an architecture. For example: A flag `sdl2` requires
+    that on the target SDL2 must be available. If not the scriptmodule will not
+    run. You can also negate a flag by putting a bang (`!`) before, thus
```

Comment:
SDL1 / SDL2 flag has no bearing on the requirements in regards to any dependencies. It's only used by runcommand and the backends module.